bumble/gatt.py

```diff
@@ -735,6 +743,24 @@ def decode_value(self, value: bytes) -> str:
         return value.decode('utf-8')
 
 
+# -----------------------------------------------------------------------------
+class SerializableCharacteristicAdapter(CharacteristicAdapter):
```

Comment:
Yes, I think it's possible to do that, and it would be nice to improve type checking. Currently, the adapter classes are bit hard to deal with when it comes to types, because they are like dynamic decorators, they are polymorphic at runtime (that's convenient to reuse the code, but not convenient for typing). I have an idea about how to fix this, by splitting the adapter classes into separate classes for attributes and attribute proxies. That should solve the problem. But that's a larger change, so I'd like to do that in a separate PR.